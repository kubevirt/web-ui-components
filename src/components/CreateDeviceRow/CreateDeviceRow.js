import React from 'react';
import PropTypes from 'prop-types';

import { ListFormFactory } from '../Form/FormFactory';
import { CancelAcceptButtons } from '../CancelAcceptButtons';

const getActions = (valid, device, LoadingComponent, onAccept, onCancel) =>
  device.creating ? (
    <LoadingComponent />
  ) : (
    <CancelAcceptButtons onAccept={onAccept} onCancel={onCancel} disabled={!valid} />
  );

export class CreateDeviceRow extends React.Component {
  state = {
    valid: false,
  };

  onFormChange = (newValue, key, valid) => {
    this.props.onChange(newValue, key);
    this.setState({
      valid,
    });
  };

  render() {
    const { device, onAccept, onCancel, LoadingComponent, deviceFields, columnSizes } = this.props;
    const actions = getActions(this.state.valid, device, LoadingComponent, onAccept, onCancel);
    return (
      <ListFormFactory
        fields={deviceFields}
        fieldsValues={device}
        actions={actions}
        onFormChange={this.onFormChange}
        columnSizes={columnSizes}
      />
    );
  }
}

CreateDeviceRow.propTypes = {
  device: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  onAccept: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  LoadingComponent: PropTypes.func.isRequired,
  deviceFields: PropTypes.object.isRequired,
  columnSizes: PropTypes.object.isRequired,
};
