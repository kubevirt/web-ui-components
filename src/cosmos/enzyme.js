import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
