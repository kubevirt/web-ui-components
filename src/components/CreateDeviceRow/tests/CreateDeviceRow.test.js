import React from 'react';
import { shallow } from 'enzyme';

import { CreateDeviceRow } from '..';

import { default as CreateDeviceRowFixture } from '../fixtures/CreateDeviceRow.fixture';

const testCreateDeviceRow = () => <CreateDeviceRow {...CreateDeviceRowFixture.props} />;

describe('<CreateDeviceRow />', () => {
  it('renders correctly', () => {
    const component = shallow(testCreateDeviceRow());
    expect(component).toMatchSnapshot();
  });
});
