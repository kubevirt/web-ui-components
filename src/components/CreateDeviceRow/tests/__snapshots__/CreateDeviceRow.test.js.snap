// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateDeviceRow /> renders correctly 1`] = `
<ListFormFactory
  actions={
    <CancelAcceptButtons
      disabled={true}
      onAccept={[Function]}
      onCancel={[Function]}
    />
  }
  columnSizes={
    Object {
      "lg": 2,
    }
  }
  fields={
    Object {
      "dropdown": Object {
        "choices": Array [
          "fooValue1",
          "fooValue",
        ],
        "defaultValue": "--- Device Dropdown ---",
        "id": "device-dropdown",
        "type": "dropdown",
      },
      "name": Object {
        "id": "device-name",
        "required": true,
        "title": "DEVICE NAME",
      },
    }
  }
  fieldsValues={
    Object {
      "dropdown": Object {
        "value": "fooValue",
      },
      "name": Object {
        "value": "fooName",
      },
    }
  }
  onFormChange={[Function]}
/>
`;
